examples/equivariant_demo.ipynb

```diff
@@ -0,0 +1,1285 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import urllib.request\n",
+    "# a collection of distorted allyl alcohol molecules from the ANI-1 dataset \n",
+    "# (see https://github.com/isayev/ANI1_dataset) with energies and forces computed using DFTB+ \n",
+    "# (see https://www.dftbplus.org/)\n",
+    "url = 'https://raw.githubusercontent.com/cosmo-epfl/librascal-example-data/833b4336a7daf471e16993158322b3ea807b9d3f/inputs/molecule_conformers_dftb.xyz'\n",
+    "# Download the file from `url`, save it in a temporary directory and get the\n",
+    "# path to it (e.g. '/tmp/tmpb48zma.txt') in the `structures_fn` variable:\n",
+    "structures_fn, headers = urllib.request.urlretrieve(url)\n",
+    "structures_fn"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Total number of structure to load\n",
+    "N = 100\n",
+    "\n",
+    "# load the structures\n",
+    "frames = read(structures_fn,':{}'.format(N))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Demonstrate the equivariance of spherical expansion coefficients"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First, we compute the density expansion coefficients on a representative dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "spherical_expansion_hypers = {\n",
+    "    \"interaction_cutoff\": 3,\n",
+    "    \"max_radial\": 8,\n",
+    "    \"max_angular\": 6,\n",
+    "    \"gaussian_sigma_constant\": 0.3,\n",
+    "    \"gaussian_sigma_type\": \"Constant\",\n",
+    "    \"cutoff_smooth_width\": 0.5,\n",
+    "    \"radial_basis\": \"GTO\",\n",
+    "}\n",
+    "\n",
+    "spex = SphericalExpansion(**spherical_expansion_hypers)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "selframe = frames[8];  sel_l = 3;    # frame and l value used for the test\n",
+    "feat_scaling = 1e6                   # just a scaling to make coefficients O(1)\n",
+    "feats = spex.transform(selframe).get_features(spex)\n",
+    "ref_feats = feat_scaling*spherical_expansion_reshape(feats, **spherical_expansion_hypers)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Then, we rotate the structure, and compute the features of the rotated configuration. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# random rotation in terms of Euler angles\n",
+    "abc = np.random.uniform(size=(3))*np.pi"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# this is the Cartesian rotation matrix (helper function, follows ZYZ convention)\n",
```

Comment:
should this be 
```suggestion
    "# this is the Cartesian rotation matrix (helper function, follows XYZ convention)\n",
```

Reply:
no, it's ZYZ. there are a gazillion conventions for the definition of Euler angles, and this is the one consistent with D matrices.